### ISSUE TITLE
Inaccurate date offset on kubernetes data

### ISSUE THREAD

**Issue by Sean-Holcomb** [kubecost/cost-analyzer-helm-chart#816]
User reports offset in data by -1 day on assets page. From user description queries grab correct data, for example in the picture below the custom date range is. from 8th March to 12th March (today). But it shows from 7th - 11th. 

![](https://user-images.githubusercontent.com/78923723/110919956-1a17f500-831d-11eb-9da5-8d9d207fa822.png)

User also notes they are in CET timezone and have the kubecostModel.utcOffset value set to "+01:00" for extended discussion of this bug up to this point see @andreb89 's comments on https://github.com/kubecost/cost-analyzer-helm-chart/issues/414

**Comment by Sean-Holcomb** [comment]
@andreb89 we have ticket for this issue and will be working on it some time soon. For now you should unset the kubecostModel.utcOffset value. If this makes a significant change for you. Please report it back here. Thank you.

**Comment by andreb89** [comment]
@Sean-Holcomb So I removed the kubecostModel.utcOffset value from the deployment. To be sure, I also completely removed the current kubecost installation and run a new fresh installation afterwards. So the good news is that now there are no "empty" days anymore. 
The dashboard now shows data for all days. The current and last day (15th and 16th) show only the internal Kubernetes costs and no out-of-Azure costs like storages, caches, etc.. The day before yesterday (14th) is missing the postgres database costs block (red blocks in the screenshot). 
So overall it's better than before. Is the missing of some costs because of the 48-hour window, that you mentioned before? Is this now the correct behavior or is still something not the way it should be?

![Screenshot from 2021-03-16 14-09-26](https://user-images.githubusercontent.com/78923723/111314260-84080580-8661-11eb-8a59-feaad291a8d1.png)

**Comment by dwbrown2** [comment]
This does seem a little longer than I would expect, but it may be normal given the average Azure billing data delays... @Sean-Holcomb can you confirm based on what you've seen so far?

Is this issue accurately titled given what we've learned?

**Comment by Sean-Holcomb** [comment]
@andreb89 There is actually no window for OOC cost, they are displayed as soon as the they are available in your export CSV,    can you rebuild your ETL and open your most recent cost export CSV and confirm for me that there are line items for the previous day while it shows no OOC costs. In the case of your last image you would be looking for line items on 2021-3-16.

On face value it looks to me like the Azure costs are still being offset by one day. but it may have to do with when Azure is releasing the CSVs and what data is in them.

The 48 window that is being talked about is exclusively for price adjustments on In-cluster costs.

**Comment by andreb89** [comment]
@Sean-Holcomb @dwbrown2 
ok, I rechecked how and when the Azure cost export is running and also the resulting csv files. I also ran a manual export this morning. So first, after the manual export and a rebuild of the asset ETL, the asset dashboard looks as follows now:
![Screenshot from 2021-03-18 09-52-57](https://user-images.githubusercontent.com/78923723/111598787-dd8e4280-87cf-11eb-88dd-3a760b94695b.png)

After comparing the exports, one from last night and one from this morning, it seems, that some objects are not included like Azure Database for PostgreSQL (the red block in the graphs), if the export runs at night. Only in the export, that I ran this morning were the costs for the PostgreSQL databases included for yesterday. But they also weren't as high as the ones from the days before, so my guess is that some product types might need a longer time until Azure can consolidate the costs and will include them in a cost export. That's currently the only conclusion I see from the behavior of the shown costs for the last 2 days, in comparison to the screenshot from the post above. This is further confirmed, that the manual cost report from this morning didn't include anything at all for today yet.

So for now, I changed the export to run in the morning at 6am and I will monitor, if for the next couple days, the shown Asset costs will stay the same and are now correctly displayed.

In the end, this might not really be a bug, just something to document. Of course, the kubecostModel.utcOffset setting might still have some problem, since this all only starting working after I removed this.

**Comment by Sean-Holcomb** [comment]
@andreb89 Ok, so it sounds like the bug here is that kubecostModel.utcOffset was not working properly. Other delays in information were caused by what was available in the Azure export CSV.

**Comment by dwbrown2** [comment]
@Sean-Holcomb can you summarize what you feel is going on here? It's not clear if this is a meaningful issue or not.

**Comment by Sean-Holcomb** [comment]
From what I can tell kubecostModel.utcOffset does not correctly display the data on the front end, however the date range of the query seemed to working correctly, just an incorrect offset is being applied when it is put into the bar chart.

**Comment by pierluigilenoci** [comment]
@dwbrown2 any news about this?

**Comment by Sean-Holcomb** [comment]
@pierluigilenoci We are still recommending that users leave this feature disabled and use UTC time. Engineering discussions about this issue are still ongoing.

**Comment by pierluigilenoci** [comment]
@dwbrown2 three month has passed, is there any news?

**Comment by kirbsauce** [comment]
Hi @pierluigilenoci , I circled back with engineering and the recommendation of UTC still stands while we re-think the offset functionality.

**Comment by Sean-Holcomb** [comment]
Closing since this feature has been deprecated by https://github.com/kubecost/cost-analyzer-helm-chart/pull/1081
